src/corners.rs

```diff
@@ -90,6 +100,140 @@ pub fn corners_fast9(image: &GrayImage, threshold: u8) -> Vec<Corner> {
     corners
 }
 
+/// A FAST corner with associated orientation as described in [Rublee, et. al.
+/// (2012)][rublee].
+///
+/// [rublee]: http://www.gwylab.com/download/ORB_2012.pdf
+#[derive(Clone, Copy, PartialEq)]
+pub struct OrientedFastCorner {
+    /// Location and FAST corner score of this corner in its associated image.
+    pub corner: Corner,
+    /// Orientation of this FAST corner as determined by computing the intensity
+    /// centroid of the local patch around the corner.
+    pub orientation: f32,
+}
+
+fn intensity_centroid(image: &GrayImage, x: u32, y: u32, radius: u32) -> f32 {
+    let mut y_centroid: i32 = 0;
+    let mut x_centroid: i32 = 0;
+
+    let (width, height) = image.dimensions();
+    let x_min = if x < radius { 0 } else { x - radius };
+    let y_min = if y < radius { 0 } else { y - radius };
+    let y_max = u32::min(y + radius + 1, height);
+    let x_max = u32::min(x + radius + 1, width);
+
+    let (mut x_count, mut y_count) = (-(radius as i32), (radius as i32));
```

Comment:
This looks wrong if `(x, y)` is within `radius + 1` of the boundary of `image`

Reply:
Like with the other problem above, I can't see the problem you've found. Can you show some values that cause this to fail? That would help me cover this case more easily.